### ISSUE TITLE
NER on custom data fails to start training and gets stuck before 1st epoch

### ISSUE THREAD

**Issue by elazzouzi1080** [flairNLP/flair#2699]
I'm trying to train a custom NER with falir using my data (BIO format).

    from flair.data import Corpus
    from flair.datasets import ColumnCorpus
    columns = {0 : 'text', 1 : 'ner'}
    data_folder = 'Data_french/'
    corpus: Corpus = ColumnCorpus(data_folder, columns,
                                  train_file = 'train.txt',
                                  test_file = 'test.txt',
                                  dev_file = 'eval.txt')

2022-04-01 13:59:25,399 Reading data from Data_french
2022-04-01 13:59:25,401 Train: Data_french/train.txt
2022-04-01 13:59:25,401 Dev: Data_french/eval.txt
2022-04-01 13:59:25,401 Test: Data_french/test.txt
CPU times: user 49min 20s, sys: 5min 54s, total: 55min 15s
Wall time: 55min 18s

    print(len(corpus.train))
    print(corpus.train[0].to_tagged_string('ner'))

4948186
Myélogramme en sortie d'aplasie le 17.04.2013 `<B-DATE>` : rémission cytologique .

    tag_type = 'ner' 
    tag_dictionary = corpus.make_label_dictionary(label_type=tag_type)

2022-04-01 15:37:39,082 Computing label dictionary. Progress:
100%|██████████| 4948186/4948186 [05:20<00:00, 15435.43it/s]
2022-04-01 15:42:59,660 Corpus contains the labels: ner (#179449386)
2022-04-01 15:42:59,660 Created (for label 'ner') Dictionary with 17 tags: <unk>, O, B-DATE, B-PATIENT, B-VILLE, I-DATE, B-DOCTOR, B-ZIP, I-PATIENT, I-DOCTOR, B-STR, I-STR, B-PHONE, I-PHONE, B-EMAIL, I-EMAIL, I-VILLE

  tag_dictionary.get_items()

[' `<unk>` ',
 'O',
 'B-DATE',
 'B-PATIENT',
 'B-VILLE',
 'I-DATE',
 'B-DOCTOR',
 'B-ZIP',
 'I-PATIENT',
 'I-DOCTOR',
 'B-STR',
 'I-STR',
 'B-PHONE',
 'I-PHONE',
 'B-EMAIL',
 'I-EMAIL',
 'I-VILLE']

      embedding_types = [
      
          # GloVe embeddings
          WordEmbeddings("fr"),
      
          # contextual string embeddings, forward
          FlairEmbeddings('fr-forward'),
      
          # contextual string embeddings, backward
          FlairEmbeddings('fr-backward'),
      ]
      embeddings : StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

      from flair.models import SequenceTagger
      tagger : SequenceTagger = SequenceTagger(hidden_size=256,
                                           embeddings=embeddings,
                                           tag_dictionary=tag_dictionary,
                                           tag_type=tag_type,
                                           use_crf=True)`

      from flair.trainers import ModelTrainer
      trainer : ModelTrainer = ModelTrainer(tagger, corpus)
      trainer.train('flair_all_data_model',
                    #train_with_dev=True,
                    learning_rate=0.1,
                    mini_batch_size=64,
                    max_epochs=150,
                   embeddings_storage_mode='none')

Now everything works find and started the training but it doesn't even reach the epoch one and stuck before the epochs in this stage:

2022-04-01 15:53:22,942 ----------------------------------------------------------------------------------------------------
2022-04-01 15:53:22,942 Model: "SequenceTagger(
  (embeddings): StackedEmbeddings(
    (list_embedding_0): WordEmbeddings(
      'fr'
      (embedding): Embedding(1000000, 300)
    )
    (list_embedding_1): FlairEmbeddings(
      (lm): LanguageModel(
        (drop): Dropout(p=0.5, inplace=False)
        (encoder): Embedding(275, 100)
        (rnn): LSTM(100, 1024)
        (decoder): Linear(in_features=1024, out_features=275, bias=True)
      )
    )
    (list_embedding_2): FlairEmbeddings(
      (lm): LanguageModel(
        (drop): Dropout(p=0.5, inplace=False)
        (encoder): Embedding(275, 100)
        (rnn): LSTM(100, 1024)
        (decoder): Linear(in_features=1024, out_features=275, bias=True)
      )
    )
  )
  (word_dropout): WordDropout(p=0.05)
  (locked_dropout): LockedDropout(p=0.5)
  (embedding2nn): Linear(in_features=2348, out_features=2348, bias=True)
  (rnn): LSTM(2348, 256, batch_first=True, bidirectional=True)
  (linear): Linear(in_features=512, out_features=19, bias=True)
  (beta): 1.0
  (weights): None
  (weight_tensor) None
)"
2022-04-01 15:53:22,943 ----------------------------------------------------------------------------------------------------
2022-04-01 15:53:22,943 Corpus: "Corpus: 4948186 train + 608305 dev + 620581 test sentences"
2022-04-01 15:53:22,943 ----------------------------------------------------------------------------------------------------
2022-04-01 15:53:22,944 Parameters:
2022-04-01 15:53:22,944  - learning_rate: "0.1"
2022-04-01 15:53:22,944  - mini_batch_size: "64"
2022-04-01 15:53:22,944  - patience: "3"
2022-04-01 15:53:22,945  - anneal_factor: "0.5"
2022-04-01 15:53:22,945  - max_epochs: "150"
2022-04-01 15:53:22,945  - shuffle: "True"
2022-04-01 15:53:22,945  - train_with_dev: "False"
2022-04-01 15:53:22,946  - batch_growth_annealing: "False"
2022-04-01 15:53:22,946 ----------------------------------------------------------------------------------------------------
2022-04-01 15:53:22,946 Model training base path: "flair_all_data_model"
2022-04-01 15:53:22,946 ----------------------------------------------------------------------------------------------------
2022-04-01 15:53:22,947 Device: cuda:0
2022-04-01 15:53:22,947 ----------------------------------------------------------------------------------------------------
2022-04-01 15:53:22,947 Embeddings storage mode: none
2022-04-01 15:53:22,949 ----------------------------------------------------------------------------------------------------`

**Comment by alanakbik** [comment]
@elazzouzi1080 I think the 'problem' is that at over 5 million sentences the dataset is too large and it takes too long. You could try downsampling by calling 

```python
corpus: Corpus = ColumnCorpus(data_folder, columns,
                              train_file = 'train.txt',
                              test_file = 'test.txt',
                              dev_file = 'eval.txt').downsample(0.01)
```

that would leave you 50k sentences which is still more than most NER datasets out there. 

Alternatively, you could downsample to 500k sentences with (downsample(0.01)) and try increasing the mini-batch size (potentially in combination with increasing the learning rate). Increasing the mini-batch size makes it faster but also requires more GPU memory.
